fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfigHelper.java

```diff
@@ -191,4 +196,57 @@ private static JsonValue getPropertyFromTenantOrDefault(String propertyName) {
         
         return result;
     }
+    
+    /**
+     * Retrieves the search property restrictions.
+     *
+     * @param resourceType the resource type
+     * @param propertyType the property type
+     * @return list of allowed values for the search property, or null if no restrictions
+     * @throws Exception
+     *             an exception
+     */
+    public static List<String> getSearchPropertyRestrictions(String resourceType, String propertyType) throws Exception {
+        String propertyField = null;
+        if (SEARCH_PROPERTY_TYPE_INCLUDE.equals(propertyType)) {
+            propertyField = FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_INCLUDES;
+        }
+        else if (SEARCH_PROPERTY_TYPE_REVINCLUDE.equals(propertyType)) {
+            propertyField = FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_REV_INCLUDES;
+        }
+
+        // Retrieve the "resources" config property group.
+        if (propertyField != null) {
+            PropertyGroup rsrcsGroup = FHIRConfigHelper.getPropertyGroup(FHIRConfiguration.PROPERTY_RESOURCES);
+            if (rsrcsGroup != null) {
+                List<PropertyEntry> rsrcsEntries = rsrcsGroup.getProperties();
+                if (rsrcsEntries != null && !rsrcsEntries.isEmpty()) {
+
+                    // Try to find search property for matching resource type
+                    for (PropertyEntry rsrcsEntry : rsrcsEntries) {
+                        if (resourceType.equals(rsrcsEntry.getName())) {
+                            PropertyGroup resourceTypeGroup = (PropertyGroup) rsrcsEntry.getValue();
+                            if (resourceTypeGroup != null) {
+                                return resourceTypeGroup.getStringListProperty(propertyField);
+                            }
+                        }
+                    }
+
+                    // Otherwise, try to find search property for "Resource" resource type
+                    for (PropertyEntry rsrcsEntry : rsrcsEntries) {
```

Comment:
Should this be in an else block? off of 223?

Reply:
I believe this is correct as is.